pkg/handler/handler.go

```diff
@@ -230,7 +250,15 @@ func (o *OpenAPIService) RegisterOpenAPIVersionedService(servePath string, handl
 					}
 
 					// serve the first matching media type in the sorted clause list
-					data, etag, lastModified := accepts.GetDataAndETag()
+					data, etag, lastModified, err := accepts.GetDataAndETag()
+					if err != nil {
+						klog.Errorf("Error in OpenAPI handler: %s", err)
+						// only return a 500 if we have no older cache data to serve
+						if data == nil {
+							w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Since we keep the old bytes, `data` can only be nil during bootstrap. I thought we could write an empty body. Or perhaps returning a 503 reflects the situation better. But I don't feel we need to return a 500. WDYT?

Reply:
I think a 503 would fit well - from what I've been able to gather, actual errors during this point in the handler's lifecycle seem to be pretty rare/borderline nonexistent, so maybe a 503 would be a good bit better, semantically-speaking. @apelisse did you have any further thoughts?

Reply:
I don't have a strong opinion. 503 sounds good to me, though again as you mentioned, hopefully no one will ever see it :-)